### PR TITLE
Added custom 404 error page

### LINKED ISSUE #440: General 404 page
Should include:

- [ ] generic header with MM logo
- [ ] dashboard and user drop-down, if user is logged in
- [ ] Message: `404: The page you requested is unavailable. Please check the URL and try again\n Maybe you were looking for one of these pages: `
- [ ] text links to the programs -- that's the content we have, might as well direct users to it
- [ ] link to report a problem: for now, mailto:mitx-support@mit.edu

### PR DESCRIPTION
**Pull request by giocalitri**
#### What are the relevant tickets?
closes #440

#### What's this PR do?
Adds a custom 404 page

#### Where should the reviewer start?
probably  `ui.views` and then '404.html'

#### How should this be manually tested?
to see the the 404 template, just go to `/404/`.
To actually test that a 404 is returned for a not existing URL, you need to:
  - set the debug to False in your `docker-compose.yml`
  - run `node node_modules/webpack/bin/webpack.js --config webpack.config.prod.js --bail` to build the static assets
  - run `docker-compose run web ./manage.py collectstatic` 

#### Any background context you want to provide?
This story does not address the case the page is not found in the react router

### CODE REVIEW

File: ui/views.py [hunk 1]
```diff
@@ -52,10 +54,39 @@ def dashboard(request, *args):  # pylint: disable=unused-argument
...
+        "dashboard.html",
         context={
             "style_src": get_bundle_url(request, "style.js"),
             "dashboard_src": get_bundle_url(request, "dashboard.js"),
             "js_settings_json": json.dumps(js_settings),
         }
     )
+
+
+def handler404(request):
```
Comment (noisecapella) [inline ui/views.py]: I feel like `handler_404` and `page_404` would be more PEP8 compliant, not sure how that treats numbers though

Comment (giocalitri) [inline ui/views.py]: I cannot change the name of `handler404`: this is the standard way to override the django default one.

Comment (giocalitri) [inline ui/views.py]: I will change the other one though

**Comment by noisecapella** [comment]
Looks great :+1:

**Pull request merged**